src/requests/actions.jl

```diff
@@ -496,6 +496,76 @@ function organize_import_block(x, _, conn)
     JSONRPC.send(conn, workspace_applyEdit_request_type, ApplyWorkspaceEditParams(missing, WorkspaceEdit(missing, TextDocumentEdit[tde])))
 end
 
+function is_string_literal(x::EXPR; inraw::Bool=false)
+    if headof(x) === :STRING # CSTParser.isstringliteral(x) # TODO: """/raw""" strings not supported yet
+        if x.parent isa EXPR && x.parent.head === :string
+            # x is part of a string with interpolation
+            return false
+        end
+        # check for literal string macro: return false for eg. foo"bar" even though "bar" is considered a string literal
+        if x.parent isa EXPR && CSTParser.ismacrocall(x.parent) && endswith(x.parent.args[1].val, "_str") &&
+            ncodeunits(x.parent.args[1].val) - ncodeunits("@_str") == x.parent.args[1].span
+            return inraw ? x.parent.args[1].val == "@raw_str" : false
+        end
+        return inraw ? false : true
+    end
+    return false
+end
+
+if isdefined(Base, :escape_raw_string)
+    using Base: escape_raw_string
+else
+    # https://github.com/JuliaLang/julia/pull/35309
+    function escape_raw_string(io, str::AbstractString)
+        escapes = 0
+        for c in str
+            if c == '\\'
+                escapes += 1
+            else
+                if c == '"'
+                    escapes = escapes * 2 + 1
+                end
+                while escapes > 0
+                    write(io, '\\')
+                    escapes -= 1
+                end
+                escapes = 0
+                write(io, c)
+            end
+        end
+        while escapes > 0
+            write(io, '\\')
+            write(io, '\\')
+            escapes -= 1
+        end
+    end
+end
+
+function convert_to_raw(x, _, conn)
+    is_string_literal(x) || return
+    file, offset = get_file_loc(x)
+    quotes = x.head === :TRIPLESTRING ? "\"\"\"" : "\"" # TODO: """ not supported yet
+    raw = "raw" * quotes * sprint(escape_raw_string, x.val) * quotes
```

Comment:
I personally like using `string` better than `*` and `valof(x)` instead of `x.val`, but doesn't really matter.